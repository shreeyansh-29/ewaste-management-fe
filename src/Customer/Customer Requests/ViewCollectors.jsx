/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React, {useEffect} from "react";
import MaterialTable from "material-table";
import {} from "@material-ui/icons";

import "../Customer.css";
import {toast} from "react-toastify";
import { TOAST_SUCCESS4 } from "../../constant/constant";
toast.configure();
export default function ViewCollectors(props) {
  const {useState} = React;
  const [btndisable, setdisable] = useState(false);

  const [columns] = useState([
    {
      title: "Name",
      field: "firstName",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "15px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "15px",
      },
    },
    {
      title: "Address",
      field: "address1",
      editable: "never",
      cellStyle: {
        textAlign: "center",
        fontSize: "15px",
      },
      headerStyle: {
        textAlign: "center",
        fontSize: "15px",
      },
    },

    {
      title: "Pincode",
      field: "pinCode",
      editable: "never",
      cellStyle: {
        fontSize: "15px",
      },
      headerStyle: {
        fontSize: "15px",
      },
    },
    {
      title: "Time",
      field: "shopTime",
      editable: "never",
      cellStyle: {
        fontSize: "15px",
      },
      headerStyle: {
        fontSize: "15px",
      },
    },
  ]);
  useEffect(() => {
    // eslint-disable-next-line react/prop-types
    var address = props.data.map(function (obj) {
      return obj.address1 + " " + obj.city + " " + obj.state;
    });

    props.data.map((obj) =>
      address.map((obj1) => {
        obj.address1 = obj1;
      })
    );
  }, []);
  const [data] = useState([...props.data]);

  const handleAccept = async (e, datas) => {
    e.preventDefault();
    const tokens = localStorage.getItem("token");
    const email = localStorage.getItem("email");
    setdisable(true);
    try {
      const response = await fetch(
        "http://localhost:8083/customer/request/dropOff",
        {
          method: "POST",
          credentials: "same-origin",
          headers: {
            "Content-Type": "application/json",
            Authorization: "Bearer " + tokens,
            EMAIL: email,
          },
          body: JSON.stringify({
            category: props.customerdata[0].category,
            itemName: props.customerdata[0].itemName,
            quantity: props.customerdata[0].quantity,
            scheduledTime: datas.shopTime,
            scheduledDate: props.customerdata[0].date,
            collectorUid: datas.uid,
          }),
        }
      );

      console.log(response);

      toast.success(TOAST_SUCCESS4, {
        position: toast.POSITION.TOP_RIGHT,
      });
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <div>
      <div style={{padding: "10px"}}>
        <h2
          style={{
            textAlign: "center",
            fontSize: "30px",
            padding: "2px,",
            color: "white",
            marginBottom: "2.5%",
            backgroundColor: " rgb(30, 28, 54)",
            borderRadius: "5px",
          }}
        >
          {" "}
          Collectors Available{" "}
        </h2>
        <MaterialTable
          title=""
          columns={columns}
          data={data}
          actions={[
            {
              icon: () => (
                <button
                  className="accept-btn"
                  disabled={btndisable}
                  style={
                    btndisable === true
                      ? {color: "red", background: "grey"}
                      : {color: "white", background: "rgb(14, 185, 207)"}
                  }
                >
                  Accept
                </button>
              ),

              onClick: (e, datas) => handleAccept(e, datas),
            },
          ]}
          options={{
            actionsColumnIndex: -1,
          }}
        />
      </div>
    </div>
  );
}
