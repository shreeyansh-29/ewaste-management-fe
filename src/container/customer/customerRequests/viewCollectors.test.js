/* eslint-disable no-undef */
import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";
import ViewCollectors from "./viewCollectors";
import MaterialTable from "material-table";
import {BrowserRouter} from "react-router-dom";

Enzyme.configure({adapter: new Adapter()});

const mockedUsedDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockedUsedDispatch,
}));

describe("View Collectors Request", () => {
  it("test ViewCollectors", () => {
    const wrapper = shallow(<ViewCollectors />);
    expect.assertions(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should test for Material Table", () => {
    const wrapper = mount(<ViewCollectors />);
    expect(wrapper.find(MaterialTable).length).toEqual(1);
  });
  it("should have accept button", () => {
    let props = {
      status: "success",
      customerdata: [
        {
          category: "Temp",
          itemName: "tv",
          quantity: "3",
        },
      ],
      data: [
        {
          city: "Mahendragarh",
          email: "coll@gmail.com",
          firstName: "Harry",
          gstNo: "07FFFFF0000B2Z9",
          id: "2",
          lastName: "Potter",
          mobileNo: "9090909987",
        },
      ],
    };
    const mockFn = jest.fn();
    const wrapper = mount(
      <BrowserRouter>
        <ViewCollectors {...props} handleAccept={mockFn} />
      </BrowserRouter>
    );
    wrapper.find("#bttn").simulate("click");
    expect(mockedUsedDispatch).toHaveBeenCalled;
  });
});
