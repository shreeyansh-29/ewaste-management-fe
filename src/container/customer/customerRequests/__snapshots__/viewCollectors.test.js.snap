// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Collectors Request test ViewCollectors 1`] = `
<div>
  <div
    style={
      Object {
        "padding": "50px 0 0 0",
      }
    }
  >
    <h2
      style={
        Object {
          "backgroundColor": " rgb(30, 28, 54)",
          "borderRadius": "5px",
          "color": "white",
          "fontSize": "30px",
          "marginBottom": "2.5%",
          "padding": "2px,",
          "textAlign": "center",
        }
      }
    >
       
      Collectors Available
       
    </h2>
    <WithStyles(Component)
      actions={
        Array [
          Object {
            "icon": [Function],
            "onClick": [Function],
          },
        ]
      }
      columns={
        Array [
          Object {
            "cellStyle": Object {
              "fontSize": "15px",
              "textAlign": "center",
            },
            "editable": "never",
            "field": "firstName",
            "headerStyle": Object {
              "fontSize": "15px",
              "textAlign": "center",
            },
            "title": "Name",
          },
          Object {
            "cellStyle": Object {
              "fontSize": "15px",
              "textAlign": "center",
            },
            "editable": "never",
            "field": "address1",
            "headerStyle": Object {
              "fontSize": "15px",
              "textAlign": "center",
            },
            "title": "Address",
          },
          Object {
            "cellStyle": Object {
              "fontSize": "15px",
            },
            "editable": "never",
            "field": "pinCode",
            "headerStyle": Object {
              "fontSize": "15px",
            },
            "title": "Pincode",
          },
          Object {
            "cellStyle": Object {
              "fontSize": "15px",
            },
            "editable": "never",
            "field": "shopTime",
            "headerStyle": Object {
              "fontSize": "15px",
            },
            "title": "Time",
          },
        ]
      }
      options={
        Object {
          "actionsColumnIndex": -1,
        }
      }
      title=""
    />
  </div>
</div>
`;
